snap/squashfs/squashfs.go

```diff
@@ -102,6 +102,32 @@ func (s *Snap) ReadFile(path string) (content []byte, err error) {
 	return ioutil.ReadFile(filepath.Join(unpackDir, path))
 }
 
+// ListDir returns the content of a single directory inside a squashfs snap.
+func (s *Snap) ListDir(path string) ([]string, error) {
+	tmpdir, err := ioutil.TempDir("", "read-dir")
+	if err != nil {
+		return nil, err
+	}
+	defer os.RemoveAll(tmpdir)
+
+	unpackDir := filepath.Join(tmpdir, "unpack")
+	if err := runCommand("unsquashfs", "-i", "-d", unpackDir, s.path, path); err != nil {
```

Comment:
@pedronis it's not terrible, but it's not as slick as what's happening now. For example, I can get output like this:

````
$ unsquashfs -no-progress -dest "" -l <snap> meta/hooks
Parallel unsquashfs: Using 8 processors
2 inodes (2 blocks) to write


/meta
/meta/hooks
/meta/hooks/install
/meta/hooks/oops_a_dir/stuff
/meta/hooks/oops_a_dir/more_stuff
/meta/hooks/upgrade
````

The parsing will need to turn this into `[]string{"install", "oops_a_dir", "upgrade"}`. It won't be beautiful, but it's doable if we think it's worth it.